### PR TITLE
Memory mapped I/O

### PR DESCRIPTION
**Pull request by LefterisJP**
Please Review:

- Adding mmap_win32.c, a wrapper of `CreateFileMapping()` for Windows to expose the good old same mmap() api we have in Posix.

- io_posix. and io_win32.c now contains the minimum amount of code possible.

- Fixed windows specific issues regarding printf format specifiers

- Tests and fixes regarding ethash_callback_t

### CODE REVIEW

File: src/libethash/ethash.h [hunk 1]
```diff
@@ -183,9 +185,11 @@ ethash_cache *ethash_light_acquire_cache(ethash_light_t light);
  * @return          Newly allocated ethash_full handler or NULL in case of
  *                  ERRNOMEM or invalid parameters used for @ref ethash_compute_full_data()
  */
-ethash_full_t ethash_full_new(ethash_params const* params,
-							  ethash_cache const* cache,
-							  ethash_callback_t callback);
+ethash_full_t ethash_full_new(char const *dirname,
+	const ethash_h256_t *seed_hash,
```
Comment (LianaHus) [inline src/libethash/ethash.h]: const*
ethash_h256_t const*

File: src/libethash/internal.c [hunk 2]
```diff
@@ -353,21 +352,51 @@ ethash_cache *ethash_light_acquire_cache(ethash_light_t light)
 	return ret;
 }
 
-ethash_full_t ethash_full_new(ethash_params const* params,
-							  ethash_cache const* cache,
-							  ethash_callback_t callback)
+ethash_full_t ethash_full_new(char const *dirname,
+	const ethash_h256_t *seed_hash,
```
Comment (LianaHus) [inline src/libethash/internal.c]: same here

Comment (LefterisJP) [inline src/libethash/internal.c]: yeah will deal with style consistency on this header in another PR thanks :)

**Comment by LefterisJP** [comment]
Travis build error is in the python code tests. Have a pretty good idea what it may be. will fix.

File: src/libethash/ethash.h (hunk 1 above)
Comment (gavofyork) [inline src/libethash/ethash.h]: +1

File: src/libethash/internal.c (hunk 2 above)
Comment (gavofyork) [inline src/libethash/internal.c]: liana is absolutely right. at an absolute minimum, you should maintain good style.

**Comment by gavofyork** [comment]
minor style fixes required; looks good otherwise.

**Pull request merged**